bucket/bucket_plain.go

```diff
@@ -5,7 +5,27 @@ import (
 )
 
 // MetricOperation is a list of metric operations to use for metric
-type MetricOperation [3]string
+type MetricOperation struct {
+	operations  [3]string
+	LabelNames  []string
+	LabelValues []string
+}
+
+// NewMetricOperation  builds and returns new MetricOperation instance with defined label keys
+func NewMetricOperation(operations [3]string, labelNames []string) MetricOperation {
```

Comment:
I'd change this to be something like this:

```
func NewMetricOperation(operaion ...string) *MetricOperation {
    ...
}
```

this makes new object creation less verbose and more elegant, and if want to labels - you already introduced `WithLabels(labels ...string)`, so you will get the following:

- now: `NewMetricOperation([3]string{"get", "token", "client_credentials"}, []string{})`
- after: `NewMetricOperation("get", "token", "client_credentials")`